src/common/process-singleton/process_singleton.h

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2017 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a MIT-style license that can be
+// found in the LICENSE file.
+
+#ifndef _PROCESS_SINGLETON_H_
+#define _PROCESS_SINGLETON_H_
+
+#include <boost/interprocess/shared_memory_object.hpp>
```

Comment:
It seems this will add boost dependency, I remember some days ago there is an issue regarding boost dependency, do we finalized the solution?